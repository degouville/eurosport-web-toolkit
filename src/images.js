import { css } from 'react-emotion';

export const retinaImage = retinaImageSrc => css`
  background-image: url(${retinaImageSrc.default});
  @media (-webkit-min-device-pixel-ratio: 2) {
    background-image: url(${retinaImageSrc.x2});
  }
  @media (-webkit-min-device-pixel-ratio: 3) {
    background-image: url(${retinaImageSrc.x3});
  }
`;

export const responsiveRetinaImage = image => css`
  ${retinaImage(image.mobile)};
  @media (min-width: 481px) {
    ${retinaImage(image.over480)};
  }
  @media (min-width: 701px) {
    ${retinaImage(image.over700)};
  }
  @media (min-width: 1025px) {
    ${retinaImage(image.over1024)};
  }
`;
