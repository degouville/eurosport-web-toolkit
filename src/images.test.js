import React from 'react';
import styled from 'react-emotion';
import { shallow } from 'enzyme';
import { retinaImage, responsiveRetinaImage } from './images';

const renderImage = image => {
  const StyledDiv = styled.div`
    background-image: ${image};
  `;
  return shallow(<StyledDiv />);
};

describe('retinaImage', () => {
  it('produces the css for a retina image', () => {
    const image = retinaImage({
      default: 'image',
      x2: 'imageX2',
      x3: 'imageX3',
    });
    expect(renderImage(image)).toMatchSnapshot();
  });
});

describe('responsiveRetinaImage', () => {
  it('produces the css for a responsive retina image', () => {
    const image = responsiveRetinaImage({
      mobile: {
        default: 'image375',
        x2: 'image375X2',
        x3: 'image375X3',
      },
      over480: {
        default: 'image768',
        x2: 'image768X2',
        x3: 'image768X3',
      },
      over700: {
        default: 'image1024',
        x2: 'image1024X2',
        x3: 'image1024X3',
      },
      over1024: {
        default: 'image1440',
        x2: 'image1440X2',
        x3: 'image1440X3',
      },
    });
    expect(renderImage(image)).toMatchSnapshot();
  });
});
