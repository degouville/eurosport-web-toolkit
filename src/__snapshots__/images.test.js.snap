// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsiveRetinaImage produces the css for a responsive retina image 1`] = `
.emotion-0 {
  background-image: background-image:url(image375);
}

@media (-webkit-min-device-pixel-ratio:2) {
  .emotion-0 {
    background-image: url(image375X2);
  }
}

@media (-webkit-min-device-pixel-ratio:3) {
  .emotion-0 {
    background-image: url(image375X3);
  }
}

@media (min-width:481px) {
  .emotion-0 {
    background-image: url(image768);
  }

@media (-webkit-min-device-pixel-ratio:2) {
    .emotion-0 {
      background-image: url(image768X2);
    }
}

@media (-webkit-min-device-pixel-ratio:3) {
    .emotion-0 {
      background-image: url(image768X3);
    }
}
}

@media (min-width:701px) {
  .emotion-0 {
    background-image: url(image1024);
  }

@media (-webkit-min-device-pixel-ratio:2) {
    .emotion-0 {
      background-image: url(image1024X2);
    }
}

@media (-webkit-min-device-pixel-ratio:3) {
    .emotion-0 {
      background-image: url(image1024X3);
    }
}
}

@media (min-width:1025px) {
  .emotion-0 {
    background-image: url(image1440);
  }

@media (-webkit-min-device-pixel-ratio:2) {
    .emotion-0 {
      background-image: url(image1440X2);
    }
}

@media (-webkit-min-device-pixel-ratio:3) {
    .emotion-0 {
      background-image: url(image1440X3);
    }
}
}

<div
  className="emotion-0 emotion-1"
/>
`;

exports[`retinaImage produces the css for a retina image 1`] = `
.emotion-0 {
  background-image: background-image:url(image);
}

@media (-webkit-min-device-pixel-ratio:2) {
  .emotion-0 {
    background-image: url(imageX2);
  }
}

@media (-webkit-min-device-pixel-ratio:3) {
  .emotion-0 {
    background-image: url(imageX3);
  }
}

<div
  className="emotion-0 emotion-1"
/>
`;
